model/context.go

```diff
@@ -107,10 +107,37 @@ func truncate(s string) string {
 	return s
 }
 
+type NetworkInfo struct {
+	EffectiveType *string
+	RoundTripTime *int64
+	Downlink      *int64
+	DownlinkMax   *int64
+	SaveData      *bool
+	Type          *string
+}
+
+func (ni *NetworkInfo) fields() common.MapStr {
+	fields := common.MapStr{}
+	if ni == nil {
+		return fields
+	}
+	utility.Set(fields, "type", ni.Type)
+	utility.Set(fields, "downlink", ni.Downlink)
+	utility.Set(fields, "downlinkMax", ni.DownlinkMax)
+	utility.Set(fields, "effectiveType", ni.EffectiveType)
+	utility.Set(fields, "roundTripTime", ni.RoundTripTime)
+	utility.Set(fields, "saveData", ni.SaveData)
+	return fields
+}
+
 // Page consists of URL and referer
 type Page struct {
-	URL     *URL
-	Referer *string
+	URL                    *URL
+	Referer                *string
+	NetworkInfo            *NetworkInfo
+	Cores                  *int
+	Memory                 *int
```

Comment:
NetworkInfo, Cores, and Memory don't really feel to me like they belong in "Page". They're properties of the device, not the page.

Reply:
In https://github.com/elastic/apm-server/issues/3657#issuecomment-644658711 I understood that these fields will be queried/used in the context of a page load, so I thought it would be intuitive to set them there, following the original proposal.  
Having a `device` field or similar on the `transaction` doesn't feel any more semantically appropriate to me (a device is not a property of a transaction), and using metricsets as I suggested in the issue would not help querying.

I guess it could be shoehorned in transaction context - just because context means pretty much anything...
Do you have any suggestion in particular?

Reply:
Good point Andrew, I am also inclined more towards having it as part of `device` that page. But would it affect/confused with the user_agent.device in any way?

Reply:
> Having a device field or similar on the transaction doesn't feel any more semantically appropriate to me (a device is not a property of a transaction), and using metricsets as I suggested in the issue would not help querying.

Fair point that it's not a property of the transaction. What I was getting at is that this information is not necessarily restricted to page-loads, so let's not overfit to this one requirement. These fields may also be applicable for serverless/lambda.

Here's a more concrete suggestion:

 1. extend our `system` model type with optional fields to report the the number of CPU cores and total memory
 2. add a `network` model type which can be reported in stream metadata
 3. allow `transaction.context.system` which should be merged with the system from stream metadata, like we do with `transaction.context.service`
 4. same for `transaction.context.network`

We could skip 3-4 if we want these properties to be associated with spans, error and metrics as well as transactions.